h_malloc.c

```diff
@@ -1874,6 +1878,86 @@ EXPORT size_t h_malloc_object_size_fast(const void *p) {
     return SIZE_MAX;
 }
 
+#if CONFIG_BLOCK_OPS_CHECK_SIZE && !defined(HAS_ARM_MTE)
+EXPORT void *memcpy(void *restrict dst, const void *restrict src, size_t len) {
+    if(dst == src || len == 0) {
```

Comment:
maybe also add `unlikely()` here? (nit: there is a missing space between `if` and `(`)